src/response/status.rs

```diff
@@ -1,178 +1,492 @@
+    /// `0x6982`
+    pub const SECURITY_STATUS_NOT_SATISFIED: Self = Self(0x6982);
+    /// `0x6983`
+    pub const AUTHENTICATION_METHOD_BLOCKED: Self = Self(0x6983);
+    /// `0x6984`
+    pub const REFERENCE_DATA_NOT_USABLE: Self = Self(0x6984);
+    /// `0x6985`
+    pub const CONDITION_OF_USE_NOT_SATISFIED: Self = Self(0x6985);
+    /// `0x6986`
+    pub const COMMAND_NOT_ALLOWED_NO_EF: Self = Self(0x6986);
+    /// `0x6987`
+    pub const EXECTED_SECURE_MESSAGING_DATA_OBJECTS_MISSING: Self = Self(0x6987);
+    /// `0x6988`
+    pub const INCORRECT_SECURE_MESSAGING_DATA_OBJECTS: Self = Self(0x6988);
+
+    /// `0x6A00`
+    pub const WRONG_PARAMETERS_NO_INFO: Self = Self(0x6A00);
+    /// `0x6A80`
+    pub const INCORRECT_PARAMETERS: Self = Self(0x6A80);
+    /// `0x6A81`
+    pub const FUNCTION_NOT_SUPPORTED: Self = Self(0x6A81);
+    /// `0x6A82`
+    pub const FILE_OR_APP_NOT_FOUND: Self = Self(0x6A82);
+    /// `0x6A83`
+    pub const RECORD_NOT_FOUND: Self = Self(0x6A83);
+    /// `0x6A84`
+    pub const NOT_ENOUGH_MEMORY_IN_FILE: Self = Self(0x6A84);
+    /// `0x6A85`
+    pub const NC_INCONSISTENT_WITH_TLV: Self = Self(0x6A85);
+    /// `0x6A86`
+    pub const INCORRECT_P1P2: Self = Self(0x6A86);
+    /// `0x6A87`
+    pub const NC_INCONSISTENT_WITH_P1P2: Self = Self(0x6A87);
+    /// `0x6A88`
+    pub const REFERENCE_NOT_FOUND: Self = Self(0x6A88);
+    /// `0x6A89`
+    pub const FILE_ALREADY_EXISTS: Self = Self(0x6A89);
+    /// `0x6A8A`
+    pub const DF_NAME_ALREADY_EXISTS: Self = Self(0x6A8A);
+
+    /// `0x6B00`
+    pub const WRONG_PARAMETERS: Self = Self(0x6B00);
+
+    const WRONG_LE_FIELD_MASK: u16 = 0x6C00;
+
+    /// `0x6D00`
+    pub const INSTRUCTION_NOT_SUPPORTED_OR_INVALID: Self = Self(0x6D00);
+    /// `0x6E00`
+    pub const CLASS_NOT_SUPPORTED: Self = Self(0x6E00);
+    /// `0x6F00`
+    pub const ERROR: Self = Self(0x6F00);
+
+    /// Create a status representing a wrong LE field (`0x6CXX`)
+    pub const fn wrong_le_field(available_bytes: u8) -> Self {
+        Self(Self::WRONG_LE_FIELD_MASK | available_bytes as u16)
+    }
 
-    ///////////////////////////////
-    // Warning processing (62, 63)
-    ///////////////////////////////
+    /// Create a status indicating that more data is available (`0x61XX`)
+    pub const fn more_available(value: u16) -> Self {
+        Self(Self::MORE_AVAILABLE_MASK | value)
+    }
 
-    // 62XX: state of non-volatile memory unchanged (cf. SW2)
+    pub const fn as_more_available(self) -> Option<u8> {
+        if self.0 | Self::MORE_AVAILABLE_MASK == Self::MORE_AVAILABLE_MASK {
+            Some((self.0 & 0x00FF) as u8)
+        } else {
+            None
+        }
+    }
+    pub const fn is_more_available(self) -> bool {
+        self.as_more_available().is_some()
+    }
 
-    // 63XX: state of non-volatile memory changed (cf. SW2)
-    VerificationFailed,
-    RemainingRetries(u8),
+    pub const fn is_warning(self) -> bool {
```

Comment:
These kind of methods are great as they expand semantic expressivity of the abstraction 👍